language/src/regent/header_helper.t

```diff
@@ -442,8 +442,10 @@ local function make_add_argument(launcher_name, wrapper_type, state_type,
     local field_paths, _ = base.types.flatten_struct_fields(param_type:fspace())
     arg_setup:insert(
       quote
-          base.assert([field_count] == [#field_paths],
-            ["wrong number of fields for region " .. tostring(arg_value) .. " (argument " .. param_i .. ")"])
+          var msg : int8[128]
+          c.sprintf(msg, [launcher_name .. " wrong number of fields for region " .. tostring(arg_value) .. " (argument " .. param_i
+                            .. ") expected: " .. #field_paths .. " got: %d"], field_count)
+          base.assert([field_count] == [#field_paths], msg)
```

Comment:
Instead of using `assert` for this, let's use an `if` to check the condition followed by `printf` and `abort` if it's false.

Reply:
This should now be fixed.